application_manager/plugins/chronos/plugin.py

```diff
@@ -0,0 +1,125 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+import threading
+import json
+import requests
+
+from application_manager.utils.ManagerChronos import *
+from application_manager.plugins.base import GenericApplicationExecutor
+from application_manager.plugins import base
+from application_manager.utils.ids import ID_Generator
+from application_manager.utils.logger import Log
+from application_manager.service import api
+
+
+
+
+LOG = Log("ChronosPlugin", "chronos_plugin.log")
+application_time_log = Log("Application_time", "application_time.log")
+
+class ChronosApplicationExecutor(GenericApplicationExecutor):
+
+    def __init__(self):
+        self.id = ID_Generator().get_ID()
+
+    def start_application(self, data):
+        try:
+            self.update_application_state("Running")
+            
+            # Credentials of framework
+            url = api.chronos_url
+            user = api.chronos_username
+            password = api.chronos_password
+            supervisor_ip = api.supervisor_ip
+
+            chronos = ManagerChronos(url, user, password)
+
+            # Obtain arguments 
+            start_job = time.time()
+            job = data['info_plugin']['job']            
+            deadline = data['info_plugin']['qos']['deadline'] + start_job 
+            job_duration = data['info_plugin']['qos']['duration'] # seconds
+            desv_deadline = data['info_plugin']['qos']['desv_deadline'] # %/100 (e.g 5% --> desv_deadline=0.05 )
+            jobname = job['name'] 
+
+            # For supervisor
+            payload = {
+                'framework': 'chronos',
+                'name': jobname,
+                'job_duration': job_duration,
+                'deadline': deadline,
+                'desv_deadline': desv_deadline,
+                'uuid': self.id,
+                'iterations': int( job['schedule'].split('/')[0][1:] )
+            }   
+
+            job = self.modify_job(supervisor_ip, jobname, job)
+            if ( chronos.sendJob(job) ):
+                self.init_webhook(supervisor_ip, payload)
+                print('Launch completed with UUID: '+ self.id)
+            else:
+                print('ERROR: Launch not completed')
+        except:
+            print "ERROR: run, forest, run!"
+    
+    def modify_job(self, api_rest_ip, jobname, job):
+        updateCommand_1 = "startedAt=$(date +%s); "
+        updateCommand_2 = "; /usr/bin/curl -H 'Content-type: application/json' -X POST " \
```

Comment:
you can break this line using ( instead of /

update_command_2 = ("something" +
                                      "something2" +
                                      "something3")